workflow/rules/05_deleterious_dfe.smk

```diff
@@ -0,0 +1,222 @@
+# Copyright 2025 Xin Huang
+#
+# GNU General Public License v3.0
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, please see
+#
+#    https://www.gnu.org/licenses/gpl-3.0.en.html
+
+import numpy as np
+import os
+
+demog_params = dadi_config["demog"]
+dfe_params = dadi_config["dfe"]
+grid_size_inference = dadi_config["grid_size"]["inference"]
+grid_size_cache = dadi_config["grid_size"]["cache"]
+gamma_pts = dadi_config["gamma_pts"]
+ratio = dadi_config["flags"]["ratio"]
+optimizations = dadi_config["flags"]["optimizations"]
+bootstrap_reps = dadi_config["flags"]["bootstrap_replicates"]
+chunk_size = dadi_config["flags"]["chunk_size"]
+
+
+rule create_pop_info:
+    input:
+        metadata=get_metadata_path,
+    output:
+        pop_info="results/dadi/{species}/dfe/{ppl}/pop.list",
+    conda:
+        "../envs/env.yaml"
+    log:
+        "logs/deleterious_dfe/create_pop_info.{species}.{ppl}.log",
+    shell:
+        """
+        grep -v captive {input.metadata} | awk -v pop={wildcards.ppl} '$2 ~ ("^" pop "$"){{print $4"\\t"$2}}' > {output.pop_info} 2> {log}
+        """
+
+
+rule generate_1pop_fs:
+    input:
+        syn_vcf="results/processed_data/{species}/1pop/{ppl}/{ppl}.biallelic.syn.snps.vcf.gz",
+        nonsyn_vcf="results/processed_data/{species}/1pop/{ppl}/{ppl}.biallelic.nonsyn.snps.vcf.gz",
+        pop_info=rules.create_pop_info.output.pop_info,
+    output:
+        syn_fs="results/dadi/{species}/dfe/{ppl}/fs/{ppl}.syn.unfolded.dadi.fs",
+        nonsyn_fs="results/dadi/{species}/dfe/{ppl}/fs/{ppl}.nonsyn.unfolded.dadi.fs",
+    conda:
+        "../envs/env.yaml"
+    log:
+        "logs/deleterious_dfe/generate_1pop_fs.{species}.{ppl}.log",
+    shell:
+        """
+        dadi-cli GenerateFs --vcf {input.syn_vcf} --pop-ids {wildcards.ppl} --pop-info {input.pop_info} --projections $(awk 'END {{print NR * 2}}' {input.pop_info}) --output {output.syn_fs} --polarized --mask-singletons 2> {log}
+        dadi-cli GenerateFs --vcf {input.nonsyn_vcf} --pop-ids {wildcards.ppl} --pop-info {input.pop_info} --projections $(awk 'END {{print NR * 2}}' {input.pop_info}) --output {output.nonsyn_fs} --polarized --mask-singletons 2>> {log}
```

Comment:
Here `--polarized`, `--mask-singletons` are flags, so we should also move them into the config, let users decide